src/SfxWeb/src/app/Models/DataModels/Replica.ts

```diff
@@ -73,13 +73,31 @@ export class ReplicaOnPartition extends DataModelBase<IRawReplicaOnPartition> {
     }
 
     public get role(): string {
+        const previousRole: string = this.raw.PreviousReplicaRole ? `Previous Role: ${this.raw.PreviousReplicaRole} | ` : '';
+
+        if (this.parent.raw.PartitionStatus === 'Reconfiguring') {
+            return `Reconfiguring - ${previousRole}Target Role: ${this.raw.ReplicaRole}`;
```

Comment:
> I believe the same logic is also in DeplolyedReplica, that will also need to be changed. In addition, let's add a unit test for this change in replica.cy.ts and deployedReplica.cy.ts

This doesn't apply to DeployedReplica, since it's using a different API call [Get-ServiceFabricDeployedReplica,
](https://learn.microsoft.com/en-us/powershell/module/servicefabric/get-servicefabricdeployedreplica?view=azureservicefabricps) and this API doesn't include the same `PreviousReplicaRole` field as [Get-ServiceFabricReplica.
](https://learn.microsoft.com/en-us/powershell/module/servicefabric/get-servicefabricreplica?view=azureservicefabricps)

> Let's change this to Reconfiguring - ${this.raw.PreviousReplicaRole} -> ${this.raw.ReplicaRole} to save space, and only show Target Role: ${this.raw.ReplicaRole} when there's no previous role.

I agree on this, thanks for the great suggestion.

Reply:
Instead of `->`, is it possible to use a right arrow symbol, e.g. `➜` (\u279c)?